SimpleInventoryManagementSystem/IDatabase.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace SimpleInventoryManagementSystem
+{
+    public interface IDatabase
+    {
+        void AddProduct(Product product);
+        void DeleteProduct(Product product);
+        void UpdateProduct(Product newProduct, string oldName);
+        Product SearchForProduct(string name);
+        void ViewAllProducts();
```

Comment:
when you talking with DB, using async/await to not blocking the thread,
also when make the methods async, for clean coding add Async suffix to methods' name